hack/run-e2e.sh

```diff
@@ -174,10 +174,18 @@ function e2e::setup_helm_server() {
     if hack::version_ge $(e2e::get_kube_version) "v1.16.0"; then
         # workaround for https://github.com/helm/helm/issues/6374
         # TODO remove this when we can upgrade to helm 2.15+, see https://github.com/helm/helm/pull/6462
-        $HELM_BIN init --service-account tiller --output yaml \
+        if [[ "$OSTYPE" == "darwin"* ]]; then
+            # mac system
+            $HELM_BIN init --service-account tiller --output yaml \
+            | sed 's@apiVersion: extensions/v1beta1@apiVersion: apps/v1@' \
+            | sed 's@  replicas: 1@  replicas: 1\'$'\n  selector: {"matchLabels": {"app": "helm", "name": "tiller"}}@' \
+            | $KUBECTL_BIN --context $KUBECONTEXT apply -f -
+        else
```

Comment:
remove the Linux branch

GNU sed on Linux is compatible with BSD sed on Mac. So we can use the same method here.

Reply:
Under the mac system, \'$'\ n is correct, the original  \n is wrong. So do  i  replace origin regex directly ?

Reply:
yes

Reply:
also add a note that `\'$'\n` is used to be compatible with BSD sed (Darwin)

Reply:
ok.